docs/contracts/functions.rst

```diff
@@ -250,33 +249,60 @@ will consume more gas than the 2300 gas stipend:
-    a function that is not available. If you want to implement the fallback
-    function only to receive ether, you should add a check
-    like ``require(msg.data.length == 0)`` to prevent invalid calls.
-
 .. warning::
     Contracts that receive Ether directly (without a function call, i.e. using ``send`` or ``transfer``)
-    but do not define a fallback function
+    but do not define a receive ether function or a payable fallback function
     throw an exception, sending back the Ether (this was different
     before Solidity v0.4.0). So if you want your contract to receive Ether,
-    you have to implement a payable fallback function.
+    you have to implement a receive ether function (using payable fallback functions for receiving Ether is not recommended).
+
 
 .. warning::
-    A contract without a payable fallback function can receive Ether as a recipient of a `coinbase transaction` (aka `miner block reward`)
+    A contract without a receive ether function can receive Ether as a recipient of a `coinbase transaction` (aka `miner block reward`)
     or as a destination of a ``selfdestruct``.
 
+    A contract cannot react to such Ether transfers and thus also cannot reject them. This is a design choice of the EVM and Solidity cannot work around it.
+
+    It also means that ``address(this).balance`` can be higher than the sum of some manual accounting implemented in a contract (i.e. having a counter updated in the receive ether function).
+
+::
+
+    pragma solidity >=0.5.0 <0.7.0;
+
+    // This contract keeps all Ether sent to it with no way
+    // to get it back.
+    contract Sink {
+        receive() external payable { }
+    }
+
+.. index:: ! fallback function, function;fallback
+
+.. _fallback-function:
+
+Fallback Function
+=================
+
+A contract can have at most one ``fallback`` function, declared using ``fallback () external [payable]``.
+This function cannot have arguments, cannot return anything and must have ``external`` visibility.
+It is executed on a call to the contract if none of the other
+functions match the given function signature, or if no data was supplied at
+all and there is no :ref:`receive ether function <receive-ether-function>`.
+The fallback function always receives data, but in order to also receive Ether
+it must be marked ``payable``.
+
+In the worst case, the fallback function can only rely on 2300 gas being
+available (see :ref:`receive ether function <receive-ether-function>` for a brief description of the implications of this).
+
+Like any function, the fallback function can execute complex operations as long as there is enough gas passed on to it.
+
+.. warning::
+    A ``payable`` fallback function is also executed for plain ether transfers, if no :ref:`receive ether function <receive-ether-function>`
+    is present. It is recommended to always define a receive ether function as well, if you define a payable fallback function.
```

Comment:
Please also add the reason for this recommendation as above.

Reply:
It is also there, this line points there.

Reply:
To which recommendation above are you referring? If the fallback function is `payable` it won't revert.